### PR TITLE
Clean up key expansion

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

**Comment by ekr** [comment]
@martinthomson @davidben This came from Valdez's suggestion. This seems super-simple but should be safe now. Can you make sure I haven't missed anything?

**Comment by martinthomson** [comment]
LGTM.  Extra layers of key separation is no longer needed now that we don't need to split the early secret into multiple keys.

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -3942,27 +3942,20 @@ keys.
...
 * The length of the key
 
 The keying material is computed using:
 
-       key = HKDF-Expand-Label(Secret,
-                               phase + ", " + purpose,
-                               "",
-                               key_length)
-
+       key = HKDF-Expand-Label(Secret, purpose, "", key_length_
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: last '_' should be a ')'

**Comment by davegarrett** [comment]
The key/iv table (next table following the edited table) is verbosity overkill. Could you just scrap that table and state that it's one of two short strings inline, instead?

**Comment by davidben** [comment]
LGTM, existing editorial comments aside.

Perhaps just write something like:

---
The traffic keying material is generated from an input traffic secret value using:

```
  [sender]_write_key = HKDF-Expand-Label(Secret, "key", "", key_length)
  [sender]_write_iv = HKDF-Expand-Label(Secret, "iv", "", iv_length)
```

[sender] denotes the sending side. The following table specifies the input traffic secret for each class of traffic keys:

[the table in this PR]

All the traffic keying material is recomputed whenever the underlying traffic secret changes (e.g., when changing from the handshake to application data keys or upon a key update).

---

This matches the `iv_length` and `[sender]_write_key` naming used elsewhere in the spec.

**Comment by ekr** [comment]
Merged by hand

**Pull request closed without merging**